test/provider/read_provider_test.dart

```diff
@@ -300,6 +303,40 @@ void main() {
       });
     });
 
+    group('starknet_getBlockWithTxs', () {
+      test('returns unimplemented method error for getStateUpdate', () async {
+        final GetBlockWithTxs response = await provider.getBlockWithTxs(
+            BlockId.blockHash(
+                blockHash: Felt.fromHexString(
+                    '0x3fbf1b9a9ed822423e87365923103a9577ebed2612afccf4c9f69c126eeeeb7')));
+        response.when(
+            error: (error) {
+              expect(error.code, equals(-32601));
+              expect(error.message,
+                  contains('method \'starknet_getBlockWithTxs\' not found'));
```

Comment:
you should be able to use the v010 pathfinder provider for this test now and expect a real value